later/lib/domain/usecases/watch_all_records_use_case.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:later/domain/repository/record_repository.dart';
+import '../models/record_model.dart';
+
+class WatchAllRecordsUseCase {
```

Comment:
To include this class into the dependency injection container, we must mark it as `@lazySingleton`. So the `injectable` package will know it should use this class.

The reason we mark it as `@lazySingleton` and not something else is because we want this to be created only once, so we'll have a _single instance_ of this class. Hence we will reuse the same instance whenever we'll do `injector<WatchAllRecordsUseCase>`.

If you have any other questions about `injectable` don't hesitate to ask or check its [documentation](https://pub.dev/packages/injectable).